rustup-init.sh

```diff
@@ -177,12 +177,23 @@ main() {
     return "$_retval"
 }
 
-check_proc() {
-    # Check for /proc by looking for the /proc/self/exe link
+get_current_exe() {
+    # Returns the executable used for system architecture detection
     # This is only run on Linux
-    if ! test -L /proc/self/exe ; then
-        err "fatal: Unable to find /proc/self/exe.  Is /proc mounted?  Installation cannot proceed without /proc."
+    local _current_exe
+    if test -L /proc/self/exe ; then
+        _current_exe=/proc/self/exe
+    else
+        echo "Warning: Unable to find /proc/self/exe.  System architecture detection might be inaccurate."
```

Comment:
Oops, a tiny typo here...

```suggestion
        echo "Warning: Unable to find /proc/self/exe. System architecture detection might be inaccurate."
```

Reply:
Done.